app/src/main/java/com/onboarding/calculator/mvp/model/CalculatorModel.java

```diff
@@ -70,6 +72,57 @@ public void setValues(String value) {
         }
     }
 
+    public void delete() {
+        if (!operator.isEmpty() && secondOperand.isEmpty()) {
+            operator = EMPTY_STRING;
+        } else
+            switch (switchOp) {
+                case FIRST_OPERAND: {
+                    if (firstOperand.length() > ZERO_INT) {
+                        firstOperand = firstOperand.substring(0, firstOperand.length() - 1);
```

Comment:
remove harcodes please, here and in the next "case"

Reply:
Solved!